Bio/SeqRecord.py

```diff
@@ -381,6 +382,82 @@ def _set_seq(self, value: Union["Seq", "MutableSeq"]) -> None:
         doc="The sequence itself, as a Seq or MutableSeq object.",
     )
 
+    @classmethod
+    def _from_validated(
+        cls,
+        seq: Optional[Union[Seq, MutableSeq]],
+        id: Optional[str],
+        name: str,
+        description: str,
+        dbxrefs: Optional[list[str]] = None,
+        features: Optional[list["SeqFeature"]] = None,
+        annotations: Optional[dict[str, Union[str, int]]] = None,
+        letter_annotations: Optional[dict[str, Sequence]] = None,
+    ) -> "SeqRecord":
+        """Faster constructor for post-validated data like copies or validated parsed data"""
+        inst = cls.__new__(cls)
+
+        inst._seq = seq
+        inst.id = id
+        inst.name = name
+        inst.description = description
+        if dbxrefs is None:
+            dbxrefs = []
+        inst.dbxrefs = dbxrefs
+        if features is None:
+            features = []
+        inst.features = features
+        if annotations is None:
+            annotations = {}
+        inst.annotations = annotations
+
+        if letter_annotations is None:
+            letter_annotations = {}
+        length = 0 if seq is None else len(seq)
+        inst._per_letter_annotations = _RestrictedDict(length=length)
+        dict.update(inst._per_letter_annotations, letter_annotations)  # type: ignore
+        return inst
+
+    def copy(self, deep: bool = True) -> "SeqRecord":
+        """Make a copy of a SeqRecord object
+
+        Arguments:
+         - deep       - whether to copy objects or references (bool)
+
+        When deep is True (default) it will make copies of objects the seq, dbxrefs, features, and values of the
+        annotations and letter_annotations dictionaries. However unlike the builtin python copy.deepcopy, this method
+        does not be applied recursively, so if the values of annotations are containers containing mutable objects,
+        those mutable objects will not be copied. However, this allows it to be significantly faster.
```

Comment:
So if you did have complex objects, should you use the builtin function? That could be more explicit.

Reply:
I agree it could be worded better. I was struggling a bit with how to word it. In 99% of cases like that, it should still be fine to use this function, because objects that deep should probably be treated like they’re immutable. You generally don’t want to be doing things like `my_seq_rec.letter_annotations[“some_annotation”][5].append(3)` even if you can. You’d at least want to rewrite like `my_seq_rec.letter_annotations[“some_annotation”][5] = new_list` . Might even be worth asking for it to be at least pseudo-immutable (ie setters turned off), because most of the code is implicitly assuming the sequences one level up are treated as immutable (otherwise lengths could change without checks occurring)

That said, the mutable/immutable distinction is a common gotcha for people not as familiar with Python, and I’d like to avoid circumstances where someone gets an obscure mutability bug and wastes a day failing to find it. This version is roughly following how Pandas defines .copy(deep=True) (except my copy is one level deeper), but I’m not sure if this is best. I also thought about setting the depth parameter as a integer =1 (shallow), 2, 3 (this depth), which would at least give a hint to the user that they might need to go deeper, but that seems like it might be a bit cryptic to the average user